### DISCUSSION TITLE
SH-ESP32 power setup with NMEA2000 and laptop USB?

### DISCUSSION THREAD

**Discussion by Unoxa** [hatlabs/discussions#117] [Q&A] [answered]
I am thinking about how best to power the SH-ESP32. In my setup I use a laptop connected to the SH-ESP32 by USB and a NMEA2000 network connected to the SH-ESP32. From the documentation using the NMEA network to power the SH-ESP32 seems not to be recommended due to lack of galvanic isolation. Is that correct? If so then power via USB or the power connector are the alternatives with the power connector having the advantage of being independent of the laptop.
Have I got this right?

**Comment by mairas** [comment, marked as answer]
You've got it right. If you have a laptop permanently connected via USB, you shouldn't power the SH-ESP32 via NMEA 2000. Although, since the laptop power supply has galvanic isolation (like all mains power supplies do), in practice you might not create a ground loop...

You can provide power via a separate power connector, but if the laptop is always connected, then I'd say it's redundant.
